### PR TITLE
[DataGrid] Improve controllable sorting

### PR DESCRIPTION
**Pull request by dtassone**
Part of #2044 (sorting)

```diff
// Props
- onSortModelChange?: (params: GridSortModelParams) => void;
+ onSortModelChange?: (model: GridSortModel) => void;

// Usage
- <DataGrid onSortModelChange = {(params: GridSortModelParams)=>  setSortModel(params.model)} />
+ <DataGrid onSortModelChange = {(model: GridSortModel)=> setSortModel(model) } />

```

### CODE REVIEW

**Comment by flaviendelangle** [comment]
https://github.com/mui-org/material-ui-x/blob/89faae80aa46309b231eb78d56f133615b425685/docs/pages/api-docs/data-grid/data-grid.md
https://github.com/mui-org/material-ui-x/blob/89faae80aa46309b231eb78d56f133615b425685/docs/pages/api-docs/data-grid/x-grid.md

Missing update on these two files for `onSortModelChange`

File: packages/grid/_modules_/grid/models/gridOptions.tsx [hunk 1]
```diff
@@ -418,9 +417,9 @@ export interface GridOptions {
   onSelectionModelChange?: (selectionModel: GridRowId[]) => void;
   /**
    * Callback fired when the sort model changes before a column is sorted.
-   * @param param With all properties from [[GridSortModelParams]].
+   * @param param With all properties from [[GridSortModel]].
    */
-  onSortModelChange?: (params: GridSortModelParams) => void;
+  onSortModelChange?: (model: GridSortModel) => void;
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/models/gridOptions.tsx:422]: We need to add the migration path to the PR's description.

**Review by m4theushw — COMMENTED** [review]
Most of the demos in https://deploy-preview-2095--material-ui-x.netlify.app/components/data-grid/sorting/ are broke or the sorting is locked.

File: docs/src/pages/components/data-grid/sorting/BasicSortingGrid.tsx [hunk 2]
```diff
@@ -9,16 +9,19 @@ export default function BasicSortingGrid() {
     maxColumns: 6,
   });
 
+  const [sortModel, setSortModel] = React.useState<GridSortModel>([
+    {
+      field: 'commodity',
+      sort: 'asc' as GridSortDirection,
```
Comment (m4theushw) [inline docs/src/pages/components/data-grid/sorting/BasicSortingGrid.tsx:15]: Why?

```suggestion
      sort: 'asc',
```

File: packages/storybook/src/stories/grid-sorting.stories.tsx [hunk 3]
```diff
@@ -455,15 +459,15 @@ export const ServerSideSorting = () => {
...
-    async (params: GridSortModelParams) => {
+    async (model: GridSortModel) => {
       setLoading(true);
-      action('onSortModelChange')(params);
+      action('onSortModelChange')(model);
 
-      const newRows = await sortServerRows(rows, params);
+      const newRows = await sortServerRows(rows, model, columns);
       setRows(newRows);
       setLoading(false);
```
Comment (m4theushw) [inline packages/storybook/src/stories/grid-sorting.stories.tsx:468]: We need to update the `sortModel` prop with the new model. It's not sorting.

**Review by m4theushw — COMMENTED** [review]
Almost there.

**Review by m4theushw — APPROVED** [review]

**Pull request merged**